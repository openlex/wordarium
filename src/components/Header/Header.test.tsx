import React from "react";
import { mount, ReactWrapper, render, shallow } from "enzyme";
import { Header } from "@components";

describe("WordListItem markup", () => {
    let user: ReactWrapper = {} as ReactWrapper;
    const userName = "Name";
    const onClickMock = jest.fn();

    beforeEach(() => {
        user = mount(<Header user="" onLogOut={onClickMock} />);
    });

    it("user block exist if user in props", () => {
        expect(user.find('[data-test-id="user"]').length).toBe(0);
    });

    it("no user block if no user", () => {
        user.setProps({
            user: userName,
        });
        expect(user.find('[data-test-id="user"]').length).toBe(1);
    });

    it("match snapshoot", () => {
        expect(
            render(
                <Header
                    user="{
            user: userName,
        }"
                    onLogOut={onClickMock}
                />
            )
        ).toMatchSnapshot();
    });
});
