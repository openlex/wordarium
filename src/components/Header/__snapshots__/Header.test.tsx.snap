// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordListItem markup match snapshoot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-srdahq-Wrapper e10k5ekn0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1cyxgdz-Title esss4xs0",
          "data-test-id": "value",
        },
        "children": Array [
          Object {
            "data": "Вордариум2",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-1mbus11-Wrapper er7bwhi0",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-vra55f-UserName-MixinFuncTextEllipse er7bwhi1",
              },
              "children": Array [
                Object {
                  "data": "{ user: userName, }",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "data-test-id": "button",
                },
                "children": Array [
                  Object {
                    "data": "Выйти",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "data-test-id": "button",
              },
              "children": Array [
                Object {
                  "data": "Выйти",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "css-vra55f-UserName-MixinFuncTextEllipse er7bwhi1",
                },
                "children": Array [
                  Object {
                    "data": "{ user: userName, }",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-1mbus11-Wrapper er7bwhi0",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-vra55f-UserName-MixinFuncTextEllipse er7bwhi1",
            },
            "children": Array [
              Object {
                "data": "{ user: userName, }",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "data-test-id": "button",
              },
              "children": Array [
                Object {
                  "data": "Выйти",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "data-test-id": "button",
            },
            "children": Array [
              Object {
                "data": "Выйти",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "css-vra55f-UserName-MixinFuncTextEllipse er7bwhi1",
              },
              "children": Array [
                Object {
                  "data": "{ user: userName, }",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-1cyxgdz-Title esss4xs0",
            "data-test-id": "value",
          },
          "children": Array [
            Object {
              "data": "Вордариум2",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
